### PR TITLE
Add argh

### PR DESCRIPTION
**Pull request by myuanz**
I didn't figure out how to run the official binary test in `on_test`, but it's a very refreshing header-only project, so if it gets compiled successfully it will run fine.

### CODE REVIEW

File: packages/a/argh/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+                cout << "\nFlags:\n";
+                for (auto& flag : cmdl.flags())
+                    cout << '\t' << flag << endl;
+
+                cout << "\nParameters:\n";
+                for (auto& param : cmdl.params())
+                    cout << '\t' << param.first << " : " << param.second << endl;
+
+                return EXIT_SUCCESS;
```
Comment (waruqi) [inline packages/a/argh/xmake.lua:44]: oh, too large test code, please simplify it.

File: packages/a/argh/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,48 @@
+package("argh")
```
Comment (waruqi) [inline packages/a/argh/xmake.lua:1]: headeroly? please add `set_kind("library", {headeronly = true})`

File: packages/a/argh/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    set_homepage("https://github.com/adishavit/argh")
+    set_description("Argh! A minimalist argument handler.")
+    set_license("BSD-3-Clause")
+
+    set_urls("https://github.com/adishavit/argh.git")
```
Comment (waruqi) [inline packages/a/argh/xmake.lua:6]:
```lua
add_urls("https://github.com/adishavit/argh/archive/refs/tags/$(version).tar.gz",
    "https://github.com/adishavit/argh.git")

set_versions("v1.3.2", "<sha256 of tar.gz>")
```

File: packages/a/argh/xmake.lua [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    add_deps("cmake")
+
+    on_install(function (package)
+        import("package.tools.cmake").install(package)
+    end)
```
Comment (waruqi) [inline packages/a/argh/xmake.lua:11]: add a newline.

File: packages/a/argh/xmake.lua [hunk 5]
```diff
@@ -0,0 +1,48 @@
+package("argh")
+    set_homepage("https://github.com/adishavit/argh")
+    set_description("Argh! A minimalist argument handler.")
+    set_license("BSD-3-Clause")
+
+    set_urls("https://github.com/adishavit/argh.git")
+    add_deps("cmake")
+
+    on_install(function (package)
+        import("package.tools.cmake").install(package)
```
Comment (waruqi) [inline packages/a/argh/xmake.lua:10]: add configs to disable test and examples

BUILD_TESTS=OFF, BUILD_EXAMPLES=OFF

or only use `os.cp("args.h", package:installdir("include"))`

File: packages/a/argh/xmake.lua (hunk 1 above)
Comment (myuanz) [reply packages/a/argh/xmake.lua:44]: I agree, but it is from the official origin example (https://github.com/adishavit/argh/blob/master/example.cpp). Do I need to remove blank lines?

Comment (SirLynix) [reply packages/a/argh/xmake.lua:44]: on_test purpose is mainly to check if the library can be properly included/used (and linked if not header only), its code is not even executed. Its role is not to test the library but to test the package has been properly built

File: packages/a/argh/xmake.lua (hunk 5 above)
Comment (myuanz) [reply packages/a/argh/xmake.lua]: It's nice to get rid of cmake

File: packages/a/argh/xmake.lua (hunk 1 above)
Comment (myuanz) [reply packages/a/argh/xmake.lua:36]: Got

**Pull request merged**